### PR TITLE
Start lookup *outside* of D when D is, e.g. an ExtensionDecl

### PR DESCRIPTION
**Pull request by davidungar**
The problem is that the wrong DeclContext is being used:

void AttributeChecker::visitCustomAttr(CustomAttr *attr) {
  auto dc = D->getInnermostDeclContext();

When D is an ExtensionDecl, it causes the lookup to start *inside* the extension. Since the source location of the attribute is *outside* the extension, the ASTScope lookup cannot find a DeclContext that matches the request.

Fix by using “getDeclContext” instead of “getInnermostDeclContext”.

Will push, and see if the PR passes all the tests.

rdar://60840456

### CODE REVIEW

**Comment by davidungar** [comment]
@swift-ci please smoke test

**Review by DougGregor — APPROVED** [review]

**Review by DougGregor — APPROVED** [review]
Test case?

**Comment by davidungar** [comment]
@swift-ci please smoke test

File: test/NameBinding/custom-attr-on-extension.swift [hunk 1]
```diff
@@ -0,0 +1,8 @@
+// Trips an assertion if an ASTScope lookup is attempted into the innermost
+// DeclContext of the extension.
+
+struct Ty{}
+
+@foo extension Ty {} // expected-error {{unknown attribute 'foo'}}
+
+// RUN: %target-swift-frontend -typecheck %s -verify
```
Comment (slavapestov) [inline test/NameBinding/custom-attr-on-extension.swift]: Typically we put the RUN: line at the top of the file

**Review by slavapestov — APPROVED** [review]

**Comment by davidungar** [comment]
@swift-ci please smoke test & merge

**Comment by davidungar** [comment]
@swift-ci please smoke test

**Pull request merged**